src/cache/sqlite_tracker.rs

```diff
@@ -129,6 +129,7 @@ impl<F> CacheTracker for SqliteCacheTracker<F>
 where
     F: Fn(i64, BlockInfo) -> Option<i64> + Send,
 {
+    #[allow(clippy::needless_collect)]
```

Comment:
why collect then?

Reply:
I tried this:
```rust
        let accessed = SystemTime::now()
            .duration_since(SystemTime::UNIX_EPOCH)
            .unwrap_or_default();
        let nanos = accessed.as_nanos() as i64;
        let mut items = blocks
            .iter()
            .filter_map(|block| (self.mk_cache_entry)(nanos, *block).map(|nanos| (block.id, nanos)))
            .peekable();
        if items.peek().is_none() {
            return;
        }
        attempt_txn(&mut self.conn, |txn| {
            for (id, accessed) in items {
                set_accessed(txn, id, accessed as i64)?;
            }
            Ok(())
        });
```

but it results in some lifetime issues

```
error[E0502]: cannot borrow `self.conn` as mutable because it is also borrowed as immutable
   --> src/cache/sqlite_tracker.rs:145:21
    |
140 |             .filter_map(|block| (self.mk_cache_entry)(nanos, *block).map(|nanos| (block.id, nanos)))
    |                         -------  ---- first borrow occurs due to use of `self` in closure
    |                         |
    |                         immutable borrow occurs here
...
145 |         attempt_txn(&mut self.conn, |txn| {
    |                     ^^^^^^^^^^^^^^ mutable borrow occurs here
146 |             for (id, accessed) in items {
    |                                   ----- immutable borrow later captured here by closure
```

I guess I could make mk_cache_entry clone to get around this, but I think collecting is fine.